funfact/parametrized.py

```diff
@@ -0,0 +1,91 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+from typing import Callable
+from funfact import active_backend as ab
+from funfact.lang._tsrex import TsrEx
+from funfact.lang._ast import Primitives as P
+from funfact.lang._terminal import ParametrizedAbstractTensor
+
+
+class Generator:
+    ''' A generator can generate a tensor from a set of parameters.
+
+    Args:
+        shape_param: Tuple(Int):
+            Shape of the parameters to generate tensor.
+        generator: Callable:
+            Function to generate tensor from parameters.
+    '''
+    def __init__(self, shape_param, generator: Callable):
+        self._shape_param = shape_param
+        self._generator = generator
+
+    def __call__(self, param):
+        return self._generator(param)
+
+    @property
+    def shape_param(self):
```

Comment:
We can just store `shape_param` as is without the `_` prefix, and then save this `@property`.